src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider.cs

```diff
@@ -67,46 +73,67 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             if (syntaxFacts.GetRefKindOfArgument(argument) != RefKind.None)
-                return;
+                return null;
         }
 
         if (semanticModel.GetSymbolInfo(invocation, cancellationToken).GetAnySymbol() is not IMethodSymbol invocationSymbol)
-            return;
+            return null;
 
         // If the user is actually passing an array to a params argument, we can't change this to be an interpolated string.
         if (invocationSymbol.IsParams())
         {
             var lastArgument = syntaxFacts.GetArgumentsOfInvocationExpression(invocation).Last();
             var lastArgumentType = semanticModel.GetTypeInfo(syntaxFacts.GetExpressionOfArgument(lastArgument), cancellationToken).Type;
             if (lastArgumentType is IArrayTypeSymbol)
-                return;
+                return null;
         }
 
         // if the user is explicitly passing in a CultureInfo, don't offer as it's likely they want specialized
         // formatting for the values.
         foreach (var argument in arguments)
         {
-            var type = semanticModel.GetTypeInfo(syntaxFacts.GetExpressionOfArgument(argument)).Type;
+            var type = semanticModel.GetTypeInfo(syntaxFacts.GetExpressionOfArgument(argument), cancellationToken).Type;
             if (type is { Name: nameof(CultureInfo), ContainingNamespace.Name: nameof(System.Globalization), ContainingNamespace.ContainingNamespace.Name: nameof(System) })
-                return;
+                return null;
         }
 
         if (ParseExpression("$" + stringToken.Text) is not TInterpolatedStringExpressionSyntax interpolatedString)
-            return;
+            return null;
 
         if (interpolatedString.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
-            return;
+            return null;
 
         var shouldReplaceInvocation = invocationSymbol is { ContainingType.SpecialType: SpecialType.System_String, Name: nameof(string.Format) };
 
+        return new(invocation, placeholderArgument, invocationSymbol, interpolatedString, shouldReplaceInvocation);
+    }
+
+    public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+    {
+        var (document, span, cancellationToken) = context;
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+        var stringType = semanticModel.Compilation.GetSpecialType(SpecialType.System_String);
+        if (stringType.IsErrorType())
+            return;
+
+        var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+        var (invocation, placeholderArgument) = await TryFindInvocationAsync().ConfigureAwait(false);
+        if (invocation is null || placeholderArgument is null)
+            return;
+
+        var data = this.AnalyzeInvocation(document, semanticModel, invocation, placeholderArgument, cancellationToken);
+        if (data is null)
+            return;
```

Comment:
the standard refactoring entrypoint has the same logic as before, but now calls into the helper to do the analysis.